app/lib/services/ecf/ecf_application_synchronization_service.rb

```diff
@@ -0,0 +1,51 @@
+module Services
+  module Ecf
+    class EcfApplicationSynchronizationService
+      def call
+        uri = build_uri
+        request = build_http_get_request(uri)
+        response = send_http_request(uri, request)
+        handle_response(response)
+      rescue StandardError => e
+        Rails.logger.error "An error occurred during application synchronization: #{e.message}"
+      end
+
+    private
+
+      def build_uri
+        URI.parse("#{ENV['ECF_APP_BASE_URL']}/api/v1/npq/application_synchronizations")
+      end
+
+      def build_http_get_request(uri)
+        request = Net::HTTP::Get.new(uri)
+        request["Authorization"] = "Bearer #{ENV['ECF_APP_BEARER_TOKEN']}"
+        request
+      end
+
+      def send_http_request(uri, request)
+        Net::HTTP.start(uri.hostname, uri.port, use_ssl: use_ssl?, read_timeout: 30) do |http|
+          http.request(request)
+        end
+      end
+
+      def handle_response(response)
+        if response.is_a?(Net::HTTPSuccess)
+          data = response_data(response)
+          data["data"].each do |record|
+            SyncApplicationStatusJob.perform_now(record)
```

Comment:
A bit of a nitpick, but I wonder if we could some additional data processing here. At the moment, this Job is performing http call, but it is also SyncApplicationStatus job responsibility to understand the structure of the response.

Could we create a generic Struct object like `Application::SynchronizationData` with clearly defined fields, parse the response onto collection fo those structs and pass it instead?

Reply:
And I would take another approach here - I personally like to have simple to follow code. Inside the loop you could have such code [inlined](https://refactoring.com/catalog/inlineFunction.html) to prevent extra layers of abstraction, as is really calling a very simple function with no logic. Also, at current stage tracking single jobs is more complicated.
```
application = record["attributes"]
lead_provider_approval_status = application["lead_provider_approval_status"]
participant_outcome_state = application["participant_outcome_state"]
application = Application.find_by!(ecf_id: application["id"])
application.update!(lead_provider_approval_status:, participant_outcome_state:)
```